### DISCUSSION TITLE
Not authorized to connect

### DISCUSSION THREAD

**Discussion by ruem2802** [rabbitmq/rabbitmq-server#5513] [Other] [answered]
1. I create new vhost in rabbitmq (name : mqtt)
2. Then a create a user (name: test)
3. I grant user (test) to vhost (mqtt)
But when i connect in mqtt: Not authorized to connect ( I only connect ok when i grant user to defautl vhost ("/") 
![image](https://user-images.githubusercontent.com/65638123/184604758-d8b6387c-a5b8-448e-b602-9b163c4f35fb.png)

![image](https://user-images.githubusercontent.com/65638123/184604486-fddece04-6090-4222-b9e6-540c9f7a713c.png)

**Comment by michaelklishin** [comment, marked as answer]
MQTT does not have a concept of a virtual host. By naming a virtual host "mqtt" you do not expose it to MQTT clients. See server logs for specific error messages but almost certainly your user simply doesn't have the permissions for the virtual host the MQTT plugin has been [configured to use](https://www.rabbitmq.com/mqtt.html#virtual-hosts).
There are two rarely used alternatives:

 * Configure a map of ports to virtual hosts for MQTT clients to connect to a specific virtual host (if it matters to them somehow)
 * Specify virtual host as part of the username using a convention specific to RabbitMQ

Both of them are [documented in a separate section of the MQTT guide](https://www.rabbitmq.com/mqtt.html#virtual-hosts).


Another relevant thing is that the seed `guest` user [can only connect from localhost](https://www.rabbitmq.com/access-control.html#loopback-users).
